pageserver/src/http/openapi_spec.yml

```diff
@@ -23,6 +23,21 @@ paths:
                   id:
                     type: integer
 
+  /v1/failpoints:
```

Comment:
I wonder, if we should document this in a public spec: ideally, I would compile this code for test build only, since allowing somebody to query the http method and break the server is odd.

Reply:
Yeah, that's fair. I wasn't sure, but figured that it was better to add it for now (at this stage of the PR) and remove later if we don't need it.

I wasn't sure whether the OpenAPI spec was used for validation of some kind (I've since checked, and I can't find anything using outside of the routes that return it). I'll remove the failpoints from the spec

Reply:
> I wasn't sure whether the OpenAPI spec was used for validation of some kind

Both, actually.
For ad-hoc validation, it is used via 
https://github.com/neondatabase/neon/blob/b8eb908a3df34f437b4f123461b14b599be4a8b4/pageserver/src/http/routes.rs#L678

yet no tests check that it's ok.

And then (via currently implicit agreement that we keep it up-to-date), we have it https://github.com/neondatabase/cloud/blob/main/swagger/pageserver.yml and some client code generation happens around it in the project.